auth/genericoauth/provider.go

```diff
@@ -28,8 +28,15 @@ func NewProvider(
 		endpoint.TokenURL = genericOAuth.TokenURL
 	}
 
+	var oauthVerifier verifier.Verifier
+	if genericOAuth.Scope != "" {
+		oauthVerifier = ScopeVerifier{scope: genericOAuth.Scope}
```

Comment:
This should probably use `NewScopeVerifier` - filling in the private field seems a little sketchy.

Or just make the field public; the constructor isn't doing much?